src/main/java/edu/kis/powp/jobs2d/command/visitor/CommandCanvasVisitor.java

```diff
@@ -0,0 +1,47 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.drivers.transformation.Point;
+
+import java.util.Iterator;
+
+public class CommandCanvasVisitor implements Visitor {
+
+    private final Canvas canvas;
+    private boolean allPointsOnCanvas = true;
+
+    public CommandCanvasVisitor(Canvas canvas) {
+        this.canvas = canvas;
+    }
+
+    @Override
+    public void visitOperateToCommand(OperateToCommand operateToCommand) {
+        Point pointToCheck = new Point(operateToCommand.getPosX(), operateToCommand.getPosY());
+        if(!canvas.checkIfPointIsOnCanvas(pointToCheck))
+            allPointsOnCanvas = false;
```

Comment:
It works, but makes visitor not reusable (once false is encountered). This is room for errors. 
It would be better to assign check result to our flag:
allPointsOnCanvas = canvas.checkIfPointIsOnCanvas(pointToCheck))
This approach requires changes in visitICompoundCommand to maintain the correct allPointsOnCanvas, but as I said it should be reusable.

Reply:
at the moment we cannot change this because many other parts of the project depend on ICompoundCommand

Reply:
OK, however it is not clear to me why.